### PR TITLE
trilium-{desktop,server}: 0.60.4 -> 0.61.14

### PR DESCRIPTION
**Pull request by eliandoran**
## Description of changes

[Link to changelog](https://github.com/zadam/trilium/releases/tag/v0.61.14)

This version can only be used if the prior installed version is 0.60.x. Not sure if we can enforce this at package level, but the 0.60.x upgrade was backported for 23.05 in https://github.com/NixOS/nixpkgs/pull/253795 since September thanks to @FliegendeWurst .

Also cleaned up some unused code.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FliegendeWurst** [comment]
> This version can only be used if the prior installed version is 0.60.x. Not sure if we can enforce this at package level, but the 0.60.x upgrade was backported for 22.05 in https://github.com/NixOS/nixpkgs/pull/253795 since September thanks to @FliegendeWurst .

We can document it in https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md#backward-incompatibilities-sec-release-2311-incompatibilities

For example, there is already: "consul has been updated to 1.16.0. [...]. Once a new Consul version has started and upgraded its data directory, it generally cannot be downgraded to the previous version." For Trilium, we also need to mention that users need to run 0.60.X at least once before upgrading to NixOS 23.11.

**Comment by eliandoran** [comment]
> > This version can only be used if the prior installed version is 0.60.x. Not sure if we can enforce this at package level, but the 0.60.x upgrade was backported for 22.05 in #253795 since September thanks to @FliegendeWurst .
> 
> We can document it in https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md#backward-incompatibilities-sec-release-2311-incompatibilities
> 
> For example, there is already: "consul has been updated to 1.16.0. [...]. Once a new Consul version has started and upgraded its data directory, it generally cannot be downgraded to the previous version." For Trilium, we also need to mention that users need to run 0.60.X at least once before upgrading to NixOS 23.11.

Really good point. Added an entry to the release note, let me know if it's clear enough.

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 1]
```diff
@@ -552,3 +552,5 @@ The module update takes care of the new config syntax and the data itself (user
...
 
 - The Linux kernel module `msr` (see [`msr(4)`](https://man7.org/linux/man-pages/man4/msr.4.html)), which provides an interface to read and write the model-specific registers (MSRs) of an x86 CPU, can now be configured via `hardware.cpu.x86.msr`.
+
+- `trilium-desktop` and `trilium-server` have been updated to [v0.61](https://github.com/zadam/trilium/releases/tag/v0.61.11). For existing installations, upgrading to this version is supported only from v0.60.X. If an older version is used, make sure to update to v0.60 (available in NixOS 22.05) first and only then to v0.61.
```
Comment (FliegendeWurst) [inline nixos/doc/manual/release-notes/rl-2311.section.md:556]:
```suggestion
- `trilium-desktop` and `trilium-server` have been updated to [v0.61](https://github.com/zadam/trilium/releases/tag/v0.61.11). For existing installations, upgrading to this version is supported only after running v0.60.X at least once. If you are still on an older version, make sure to update to v0.60 (available in NixOS 23.05) first and only then to v0.61 (available in NixOS 23.11).
```
It's 2023 :p

**Review by FliegendeWurst — COMMENTED** [review]
The rl-notes entry should be in {#sec-release-23.11-incompatibilities}, not nixpkgs internals.

Besides that it seems like 0.61.12 is about to be released: https://github.com/zadam/trilium/issues/4382#issuecomment-1793232063
Edit: and there are open bug reports for 0.61.11 that sound scary (https://github.com/zadam/trilium/issues/4379)

File: nixos/doc/manual/release-notes/rl-2311.section.md (hunk 1 above)
Comment (eliandoran) [reply nixos/doc/manual/release-notes/rl-2311.section.md]: Ha, must have been tired.

I've updated the release note and updated to 0.61.12.

Comment (FliegendeWurst) [reply nixos/doc/manual/release-notes/rl-2311.section.md]: 0.61.13 is out: https://github.com/zadam/trilium/releases/tag/v0.61.13 with a fix for the

> sync error with revisions (the underlying cause is unknown, but a more graceful error handling should mitigate the problem)

Comment (eliandoran) [reply nixos/doc/manual/release-notes/rl-2311.section.md]: @FliegendeWurst , done.

**Review by FliegendeWurst — APPROVED** [review]
Tested by cherry-picking on NixOS 23.05.

- [x] {desktop,server} build
- [x] migration runs with no errors
- [x] {desktop,server} work

**Comment by FliegendeWurst** [comment]
By the way, it was recently suggested that we switch to nixpkgs' electron (instead of patching the binary). I think we can attempt that in another PR once 23.11 is branched off.

**Comment by eliandoran** [comment]
> By the way, it was recently suggested that we switch to nixpkgs' electron (instead of patching the binary). I think we can attempt that in another PR once 23.11 is branched off.

@FliegendeWurst , sure, we can give that a try. Do you have more info on the subject?

**Comment by FliegendeWurst** [comment]
> > By the way, it was recently suggested that we switch to nixpkgs' electron (instead of patching the binary). I think we can attempt that in another PR once 23.11 is branched off.
> 
> @FliegendeWurst , sure, we can give that a try. Do you have more info on the subject?

Most recently this sort of change was done in #266084.

**Comment by FliegendeWurst** [comment]
Please fix the merge conflict. I suspect the conflict is in the rl-notes.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by FliegendeWurst** [comment]
Another merge conflict in the rl notes?
And 0.61.14 was released a few hours ago.

**Comment by eliandoran** [comment]
> Another merge conflict in the rl notes?
> 
> 0.61.14 was released a few hours ago with some important bugfixes.

Done.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**